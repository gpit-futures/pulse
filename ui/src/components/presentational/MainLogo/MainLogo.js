import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';

export default class MainLogo extends PureComponent {
  render() {
    const { patientsInfo, userAccount, logo } = this.props;
    return (
      <div className="wrap-logo">
        <div className="logo">
          <div className="logo-icon">
            {userAccount.role === 'IDCR'
              ? <Link to={'/'}>
                {logo ? <img className="img" alt="logo" src={logo} /> : null}
                <img className={classNames(`img logo-img ${logo ? 'hidden' : ''}`)} alt="logo" src={patientsInfo.logoB64} />
              </Link>
              : userAccount.role === 'PHR'
                ? <Link to={`/patients/${userAccount.nhsNumber}/patients-summary`}>
                  {logo ? <img className="img" alt="logo" src={logo} /> : null}
                  <img className={classNames(`img logo-img ${logo ? 'hidden' : ''}`)} alt="logo" src={patientsInfo.logoB64} />
                </Link> : null }
          </div>
        </div>
      </div>
    )
  }
}
