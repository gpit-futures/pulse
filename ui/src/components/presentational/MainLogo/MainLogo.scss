.wrap-logo {
  float: left;
  margin: auto;
  margin-left: 0;
  padding: 6px 9px;
}
.logo {
  font-size: 0;
  line-height: 0;
  user-select: none;
  .logo-icon {
    cursor: pointer;
    display: inline-block;
    height: 38px;
    vertical-align: middle;
    width: auto;
    @include respond-to(xs) {
      display: flex;
      flex-direction: column;
      justify-content: center;
    }
    .img {
      display: block;
      width: 100px;
      height: auto;
      @include respond-from(xs) {
        height: 38px;
        width: auto;
      }
    }
  }
}

@include respond-from(sm) {
  .wrap-logo {
    order: 2;
    float: none;
    padding-left: 9px;
    padding-right: 9px;
  }
}
