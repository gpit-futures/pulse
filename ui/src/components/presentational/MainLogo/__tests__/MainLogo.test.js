import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import MainLogo from '../MainLogo';

Enzyme.configure({ adapter: new Adapter() });

const patientsInfo = {
  'browserTitle': 'PulseTile',
  'logoB64': 'testLogo',
  'themeColor': 'green',
  'title': 'PulseTile',
};

describe('Component <MainLogo />', () => {
  it('should renders when the role of doctor', () => {
    const userAccount = { 'role': 'IDCR' };

    const component = shallow(
      <MainLogo
        patientsInfo={patientsInfo}
        userAccount={userAccount}
      />);

    expect(component.find('Link')).toHaveLength(1);
    expect(component.find('Link').props().to).toEqual('/');
    expect(component.find('img').props().className).toEqual('img logo-img ');

    expect(component).toMatchSnapshot();

    component.setProps({ logo: 'test' });
    expect(component.find('img').at(0).props().src).toEqual('test');
    expect(component.find('img').at(1).props().className).toEqual('img logo-img hidden');

    expect(component).toMatchSnapshot();
  });

  it('should renders when the role of the patient', () => {
    const userAccount = { 'role': 'PHR', 'nhsNumber': '9999999000' };
    const component = shallow(
      <MainLogo
        patientsInfo={patientsInfo}
        userAccount={userAccount}
      />);

    expect(component.find('Link')).toHaveLength(1);
    expect(component.find('Link').props().to).toEqual('/patients/9999999000/patients-summary');
    expect(component.find('img').props().className).toEqual('img logo-img ');

    expect(component).toMatchSnapshot();

    component.setProps({ logo: 'test' });
    expect(component.find('img').at(0).props().src).toEqual('test');
    expect(component.find('img').at(1).props().className).toEqual('img logo-img hidden');

    expect(component).toMatchSnapshot();
  });

  it('should renders without role', () => {
    const userAccount = { 'role': '', 'nhsNumber': '9999999000' };
    const component = shallow(
      <MainLogo
        patientsInfo={patientsInfo}
        userAccount={userAccount}
      />);

    expect(component.find('Link')).toHaveLength(0);
    expect(component.find('img')).toHaveLength(0);

    expect(component).toMatchSnapshot();
  });
});
