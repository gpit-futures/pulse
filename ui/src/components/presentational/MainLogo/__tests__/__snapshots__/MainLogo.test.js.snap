// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <MainLogo /> should renders when the role of doctor 1`] = `
<div
  className="wrap-logo"
>
  <div
    className="logo"
  >
    <div
      className="logo-icon"
    >
      <Link
        replace={false}
        to="/"
      >
        <img
          alt="logo"
          className="img logo-img "
          src="testLogo"
        />
      </Link>
    </div>
  </div>
</div>
`;

exports[`Component <MainLogo /> should renders when the role of doctor 2`] = `
<div
  className="wrap-logo"
>
  <div
    className="logo"
  >
    <div
      className="logo-icon"
    >
      <Link
        replace={false}
        to="/"
      >
        <img
          alt="logo"
          className="img"
          src="test"
        />
        <img
          alt="logo"
          className="img logo-img hidden"
          src="testLogo"
        />
      </Link>
    </div>
  </div>
</div>
`;

exports[`Component <MainLogo /> should renders when the role of the patient 1`] = `
<div
  className="wrap-logo"
>
  <div
    className="logo"
  >
    <div
      className="logo-icon"
    >
      <Link
        replace={false}
        to="/patients/9999999000/patients-summary"
      >
        <img
          alt="logo"
          className="img logo-img "
          src="testLogo"
        />
      </Link>
    </div>
  </div>
</div>
`;

exports[`Component <MainLogo /> should renders when the role of the patient 2`] = `
<div
  className="wrap-logo"
>
  <div
    className="logo"
  >
    <div
      className="logo-icon"
    >
      <Link
        replace={false}
        to="/patients/9999999000/patients-summary"
      >
        <img
          alt="logo"
          className="img"
          src="test"
        />
        <img
          alt="logo"
          className="img logo-img hidden"
          src="testLogo"
        />
      </Link>
    </div>
  </div>
</div>
`;

exports[`Component <MainLogo /> should renders without role 1`] = `
<div
  className="wrap-logo"
>
  <div
    className="logo"
  >
    <div
      className="logo-icon"
    />
  </div>
</div>
`;
